text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/TextColorSerializer.java

```diff
@@ -49,10 +49,14 @@ public void write(final JsonWriter out, final TextColor value) throws IOExceptio
     } else if (this.downsampleColor) {
       out.value(NamedTextColor.NAMES.key(NamedTextColor.nearestTo(value)));
     } else {
-      out.value(value.asHexString());
+      out.value(asUpperCaseHexString(value));
     }
   }
 
+  private static String asUpperCaseHexString(final TextColor color) {
+    return String.format("#%06X", color.value()); // to be consistent with vanilla
```

Comment:
```suggestion
    return String.format(Locale.ROOT, "#%06X", color.value()); // to be consistent with vanilla
```

just in case number formatting is locale-dependent